File: examples/c++/ms5611.cxx
/*
 * Author: Henry Bruce <henry.bruce@intel.com>
 * Copyright (c) 2016 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */

#include <iostream>
#include <signal.h>
#include <stdio.h>

#include "ms5611.hpp"
#include "upm_utilities.h"

static int is_running = 0;

void
sig_handler(int signo)
{
    printf("got signal\n");
    if (signo == SIGINT) {
        is_running = 1;
    }
}

//! [Interesting]
int
main(int argc, char** argv)
{
    upm::MS5611 sensor(0);
    signal(SIGINT, sig_handler);

    while (!is_running) {
        int value = sensor.getTemperatureCelsius();
        std::cout << "Temperature = " << value << "C" << std::endl;
        value = sensor.getPressurePa();
        std::cout << "Pressure = " << value << "Pa" << std::endl;
        upm_delay(1);
    }

    std::cout << "exiting application" << std::endl;

    return 0;
}
//! [Interesting]

File: examples/javascript/ms5611.js
/*
 * Author: Henry Bruce <henry.bruce@intel.com>
 * Copyright (c) 2016 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */

var factory = require('jsupm_ms5611');

// Instantiate a MS5611 instance using default i2c bus and address
var sensor = new factory.MS5611();

// now output data every 250 milliseconds
setInterval(function()
{
    var temperature = sensor.getTemperatureCelsius();
    var pressure = sensor.getPressurePa();
    console.log("Temperature: " + temperature + " C");
    console.log("Pressure: " + pressure + " Pa");
}, 250);

// exit on ^C
process.on('SIGINT', function()
{
    sensor = null;
    sensorObj.cleanUp();
    sensorObj = null;
    console.log("Exiting.");
    process.exit(0);
});
